### PR TITLE
Change parameter to ?string for all methods which are dealing

### PR DESCRIPTION
**Pull request by usox**
...with uri parts #12

### CODE REVIEW

File: src/UriInterface.hh [hunk 1]
```diff
@@ -215,7 +214,7 @@ interface UriInterface {
    * an instance that contains the specified user information.
    *
    * Password is optional, but the user information MUST include the
-   * user; an empty string for the user is equivalent to removing user
+   * user; a null value for the user is equivalent to removing user
    * information.
    */
   public function withUserInfo(string $user, ?string $password = null): this;
```
Comment (fredemmott) [inline src/UriInterface.hh]: you're not allowing a null user

Comment (usox) [reply src/UriInterface.hh]: Right, that's not optimal...

**Comment by usox** [comment]
The failed build seems to be related to internal problems in the hhvm nightly...

**Comment by fredemmott** [comment]
- nightly should be fixed
- please rebase and resubmit, given the license change

**Comment by usox** [comment]
Done

**Pull request merged**